src/evalpoly.jl

```diff
@@ -0,0 +1,42 @@
+# Copyright (c) 2023 MutableArithmetics.jl contributors
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v.2.0. If a copy of the MPL was not distributed with this file, You can obtain
+# one at http://mozilla.org/MPL/2.0/.
+
+# `Base.evalpoly` MA operation implemented using Horner's scheme. For
+# real-valued polynomials with real coefficients.
+
+function promote_operation(
+    ::typeof(Base.evalpoly),
+    ::Type{F},
+    ::Type{<:TupleOrVector{F}},
+) where {F<:Real}
+    return F
+end
+
+struct PolynomialWithoutCoefficientsException <: Exception end
+
+function operate_to!(
+    val::F,
+    ::typeof(Base.evalpoly),
+    x::F,
+    coefs::TupleOrVector{F},
+) where {F<:Real}
+    # The coefficient vector is empty:
+    #
+    # Arguably a polynomial with no coefficients is the zero-polynomial, but
+    # let's follow Base.evalpoly instead of extending it.
```

Comment:
In `Base`, they can't handle the empty case easily because they don't want to infer the return type. In our case, we can just return `val` after `operate!(zero, val)` so let's do that